### ISSUE TITLE
unknown chunks in mdx, written mdx cannot be parsed by mdlx

### ISSUE THREAD

**Issue by Frotty** [inwc3/wc3libs#76]
[farmland7.zip](https://github.com/inwc3/wc3libs/files/9096725/farmland7.zip)

I'm trying to read and write the attached .mdx
I get unknown chunk errors when reading and after writing the resulting mdx is broken and cannot be parsed by mdlx.

```kotlin
    @Test
    fun assertProps() {
        val file = ClassPathResource("farmland7.mdx").file
        val mdxBytes = Files.readAllBytes(file.toPath())
        ByteArrayInputStream (mdxBytes).use {
            Wc3BinInputStream(it).use { inputStream ->
                val mdx = MDX(inputStream)
                val byteArrayOutputStream = ByteArrayOutputStream()
                val output = Wc3BinOutputStream(byteArrayOutputStream)
                mdx.write(output)

                output.close()
                val byteArrayOut = byteArrayOutputStream.toByteArray()

                Files.write(Paths.get("out.mdx"), byteArrayOut)

                Assertions.assertNotEquals(mdxBytes, byteArrayOut)
            }
        }
    }
```

[out.zip](https://github.com/inwc3/wc3libs/files/9096735/out.zip)

**Comment by Frotty** [comment]
[farmland7.zip](https://github.com/inwc3/wc3libs/files/9104205/farmland7.zip)
mdl version

**Comment by WaterKnight** [comment]
resolved in 64e514da962cfa9d353cb61abc670726608cb798, closing here
